### PR TITLE
test: use consumer/producer in servicebus integration tests

### PR DESCRIPTION
**Pull request by stijnmoreels**
Use same consumer/producer approach to integration test Service Bus functionality as in Messaging.

### CODE REVIEW

File: src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs [hunk 1]
```diff
@@ -87,12 +88,12 @@ public DirectoryInfo GetServiceBusProjectDirectory(ServiceBusEntity entity)
         /// Gets the project directory of the Azure Functions Service Bus project based on the given <paramref name="entity"/>.
         /// </summary>
         /// <exception cref="ArgumentOutOfRangeException">Thrown when no project directory can be found for the given <paramref name="entity"/>.</exception>
-        public DirectoryInfo GetAzureFunctionsServiceBusProjectDirectory(ServiceBusEntity entity)
+        public DirectoryInfo GetAzureFunctionsServiceBusProjectDirectory(ServiceBusEntityType entity)
```
Comment (fgheysels) [inline src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs:91]: Minor: should we rename the parameter as well ?

Comment (fgheysels) [inline src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs:91]:
```suggestion
        public DirectoryInfo GetAzureFunctionsServiceBusProjectDirectory(ServiceBusEntityType entityType)
```

File: src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs [hunk 2]
```diff
@@ -87,12 +88,12 @@ public DirectoryInfo GetServiceBusProjectDirectory(ServiceBusEntity entity)
...
         {
             switch (entity)
```
Comment (fgheysels) [inline src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs:93]:
```suggestion
            switch (entityType)
```

File: src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs [hunk 3]
```diff
@@ -87,12 +88,12 @@ public DirectoryInfo GetServiceBusProjectDirectory(ServiceBusEntity entity)
         /// Gets the project directory of the Azure Functions Service Bus project based on the given <paramref name="entity"/>.
         /// </summary>
         /// <exception cref="ArgumentOutOfRangeException">Thrown when no project directory can be found for the given <paramref name="entity"/>.</exception>
```
Comment (fgheysels) [inline src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs:90]:
```suggestion
        /// <exception cref="ArgumentOutOfRangeException">Thrown when no project directory can be found for the given <paramref name="entityType"/>.</exception>
```

File: src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs [hunk 4]
```diff
@@ -87,12 +88,12 @@ public DirectoryInfo GetServiceBusProjectDirectory(ServiceBusEntity entity)
         /// Gets the project directory of the Azure Functions Service Bus project based on the given <paramref name="entity"/>.
```
Comment (fgheysels) [inline src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs:88]:
```suggestion
        /// Gets the project directory of the Azure Functions Service Bus project based on the given <paramref name="entityType"/>.
```

File: src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs [hunk 5]
```diff
@@ -87,12 +88,12 @@ public DirectoryInfo GetServiceBusProjectDirectory(ServiceBusEntity entity)
...
+        public DirectoryInfo GetAzureFunctionsServiceBusProjectDirectory(ServiceBusEntityType entity)
         {
             switch (entity)
             {
-                case ServiceBusEntity.Queue: return PathCombineWithSourcesDirectory("Arcus.Templates.AzureFunctions.ServiceBus.Queue");
-                case ServiceBusEntity.Topic: return PathCombineWithSourcesDirectory("Arcus.Templates.AzureFunctions.ServiceBus.Topic");
+                case ServiceBusEntityType.Queue: return PathCombineWithSourcesDirectory("Arcus.Templates.AzureFunctions.ServiceBus.Queue");
+                case ServiceBusEntityType.Topic: return PathCombineWithSourcesDirectory("Arcus.Templates.AzureFunctions.ServiceBus.Topic");
                 default:
                     throw new ArgumentOutOfRangeException(nameof(entity), entity, "Unknown Service Bus entity");
```
Comment (fgheysels) [inline src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs:98]:
```suggestion
                    throw new ArgumentOutOfRangeException(nameof(entityType), entityType, "Unknown Service Bus entity type");
```

File: src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs [hunk 6]
```diff
@@ -221,12 +222,12 @@ public int GetDockerServiceBusTopicWorkerHealthPort()
         /// </summary>
         /// <param name="entity"></param>
         /// <returns></returns>
-        public string GetServiceBusConnectionString(ServiceBusEntity entity)
+        public string GetServiceBusConnectionString(ServiceBusEntityType entity)
```
Comment (fgheysels) [inline src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs:225]: Same here, should we rename the parameter as well ?

**Review by fgheysels — APPROVED** [review]

File: src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs (hunk 1 above)
Comment (stijnmoreels) [reply src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs]: Yes, done!

File: src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs (hunk 6 above)
Comment (stijnmoreels) [reply src/Arcus.Templates.Tests.Integration/Fixture/TestConfig.cs]: Yes, done!

**Pull request merged**